hardware-discovery-agent/internal/gpu/gpu.go

```diff
@@ -48,18 +52,24 @@ func GetGpuList(executor utils.CmdExecutor) ([]*Gpu, error) {
 		deviceFeatures := parseGpuDetails(gpu, "capabilities: ")
 		features := strings.Split(deviceFeatures[0], " ")
 
-		deviceInfo, err := utils.ReadFromCommand(executor, "lspci", "-v", "-s", pciAddress)
-		if err != nil {
-			return []*Gpu{}, fmt.Errorf("failed to read data from command; error: %v", err)
+		device := ""
```

Comment:
shall we return an error if pciAdress is empty ? Or is there a way for us to get the device name in case of non-pci devices?

Reply:
No, usually if the agent can't find some information for a HW device, it just sets that value as blank and returns what can be read, so there is at least partial information available that can be viewed. Returning an error here would prevent that and, since it is returning a series of devices, could prevent other devices listed after the one without a PCI address from being reported.

No, the only way to get the device name is to check its PCI address

Reply:
maybe default to "Info N/A" or something similar?

Reply:
Updated to `Info Not Available` for the device name, for the other components the agent will now return `Not Available` for those if they are not found